### PR TITLE
Test to snapshot based mirroring of cloned images

### PR DESCRIPTION
**Pull request by sunilangadi2**
# Description
Fixes https://issues.redhat.com/browse/RHCEPHQE-10800
Automation of test case
https://polarion.engineering.redhat.com/polarion/#/project/CEPH/workitem?id=CEPH-83576099

Test result [passed]
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-4EXAAS

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

File: tests/rbd_mirror/test_rbd_mirror_cloned_image.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+            snap_name=snap_name,
+            pool_name=poolname,
+            image_name=clone,
+        )
+
+        # Tried to enable cloned images for snapshot-based mirroring it should not allow
+        command = f"rbd mirror image enable {poolname}/{clone} snapshot"
+        out, err = rbd1.exec_cmd(cmd=command, all=True, check_ec=False)
+        if "not supported" not in err:
+            log.info(
```
Comment (manasagowri) [inline tests/rbd_mirror/test_rbd_mirror_cloned_image.py:72]: This should be log.error

File: tests/rbd_mirror/test_rbd_mirror_cloned_image.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+            "cloned image has parent, hence it's snapshot-based mirroring is not supported"
+        )
+
+        # Flatten the cloned image
+        rbd1.flatten_clone(poolname, clone)
+
+        # Enable snapshot based mirroring for the flattened child image
+        out, err = rbd1.exec_cmd(cmd=command, all=True, check_ec=False)
+        if "Mirroring enabled" not in out:
+            log.info(
```
Comment (manasagowri) [inline tests/rbd_mirror/test_rbd_mirror_cloned_image.py:87]: This should be log.error

File: tests/rbd_mirror/test_rbd_mirror_cloned_image.py (hunk 1 above)
Comment (sunilangadi2) [reply tests/rbd_mirror/test_rbd_mirror_cloned_image.py]: Done.

File: tests/rbd_mirror/test_rbd_mirror_cloned_image.py (hunk 2 above)
Comment (sunilangadi2) [reply tests/rbd_mirror/test_rbd_mirror_cloned_image.py]: Done.

**Comment by sunilangadi2** [comment]
Thanks @manasagowri for the review addressed review comments,
Please re-review it.

**Review by manasagowri — APPROVED** [review]
LGTM

**Comment by sunilangadi2** [comment]
@manasagowri this fix applicable only for quincy and reef hence removed this test from pacific.

**Pull request merged**